src/containers/PaymentFinalizationHandler/PaymentFinalizationHandler.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PaymentFinalizationPage from 'components/PaymentFinalizationPage';
+import { setShouldShowFinalizePaymentComponent } from 'redux/finalizePaymentSlice';
+
+const PaymentFinalizationHandler = Component => {
```

Comment:
maybe, because it is HOC, we could change the name to sth like `withPaymentFinalizationHandler`?